google/cloud/bigtable/instance_admin_new_test.cc

```diff
@@ -12,85 +12,94 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "google/cloud/bigtable/admin/mocks/mock_bigtable_instance_admin_connection.h"
 #include "google/cloud/bigtable/instance_admin.h"
-#include "google/cloud/bigtable/testing/mock_instance_admin_client.h"
 #include <gmock/gmock.h>
 
 namespace google {
 namespace cloud {
 namespace bigtable {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
-namespace {
 
-using ::testing::ReturnRef;
+// Helper class for checking that the legacy API still functions correctly
+class InstanceAdminTester {
```

Comment:
No action needed, just for you to think about: this is a "test smell", you need this to verify that the `private` sections of a class behave a certain way, the question should be: what behavior of the class do I need to test?  Is that behavior observable?  If not, why do I want a test? (sometimes the answer is: the behavior is only observable in terms of performance and testing for that is difficult, so this may be Okay... ).

Reply:
Hmm. (I think) I want to test that sharing one `InstanceAdminClient` across multiple `InstanceAdmin`s only shares one `BigtableInstanceAdminConnection` under the hood.
https://github.com/googleapis/google-cloud-cpp/blob/be590f9306c7fde14b906791eb751da256129e21/google/cloud/bigtable/instance_admin_new_test.cc#L93-L101

If we didn't do this, maybe it takes longer to make new `InstanceAdmin`s then it should. Or maybe we start using too many threads or something. So I think it is like you say: there is an observable difference, but it is not easy to test for it.